test/renderer_test.rb

```diff
@@ -65,6 +65,24 @@ class RendererTest < NicePartials::Test
     assert_equal "Some extra content", nice_partial.yield
   end
 
+  test "mixing Partial#yield call styles renders all captured content" do
+    render template: "mixed_yield_test"
+
+    assert_css "#mixed_yield" do
+      assert_css "#output_buffer", text: "output buffer content"
+      assert_css "#slot", text: "slot content"
+    end
+  end
```

Comment:
It's worth noting that this change passes this test:

```diff
--- a/test/fixtures/yields/_mixed.html.erb
+++ b/test/fixtures/yields/_mixed.html.erb
@@ -1,5 +1,5 @@
 <div id="mixed_yield">
   <%= partial.yield :slot %>
 
-  <%= partial.yield %>
+  <%= partial.output_buffer %>
 </div>
```

Reply:
Ah, so we aren't properly routing the call to `output_buffer` when `yield` without arguments is called?

Reply:
@kaspth I've found it to be related to the `has_capturing_yield?` check with `\byield[\(? ]+(%>|[^:])`.

The "working" partial [passes the regular expression](https://rubular.com/r/qH0ZY8hY33LPEB), which does not detect a bare `yield` call:

```erb
<div id="mixed_yield">
  <%= partial.yield :slot %>
  <%= partial.output_buffer %>
</div>
```

A partial with a bare `yield` call [does not pass the regular expression](https://rubular.com/r/DwK3LcyrQtRw2m), since it detects the bare `yield` call:

```erb
<div id="mixed_yield">
  <%= partial.yield :slot %>
  <%= yield %>
</div>
```

Unfortunately, a partial with `partial.yield` [also does not pass the regular expression](https://rubular.com/r/gJ4Qu7z7eBBAq4):

```erb
<div id="mixed_yield">
  <%= partial.yield :slot %>
  <%= partial.yield %>
</div>
```

The `\b` word boundary in the regular expression isn't treating `partial.yield` any differently than `yield`.